### DISCUSSION TITLE
Failing to remove DRM from comic Nook ePubs

### DISCUSSION THREAD

**Discussion by dmunsil** [noDRM/DeDRM_tools#143] [Q&A]
Hi All,

I'm trying to remove DRM from a graphic novel that is only available in eBook form from B&N, so I've purchased it on Nook. I have the Nook for Windows app (current version), and have successfully extracted the key. General-purpose books like novels and so forth work fine. But I have two comic format books in my library, and both of them fail to decrypt. Could it be that the extraction script expects there to be some text in the book? With a comic or graphic novel, it's just a sequence of images.

I can provide logs, metadata files, etc.

Here's a free, reasonably short comic available on B&N that repros the issue: Amazing Agent Luna #1.

https://www.barnesandnoble.com/w/amazing-agent-luna-volume-1-nunzio-defilippis/1025120194?ean=9781935934202

I haven't actually tried setting up a debugger, but just reading the code and looking at these files, it looks like the culprit could be the code that gets the encryptedKey from rights.xml, because these comic files don't have a rights.xml. Not sure why. Perhaps the JPG files are just encrypted with the base key?

https://github.com/noDRM/DeDRM_tools/blob/21281baf21389c7f1c93321c64be0ec3715a0016/DeDRM_plugin/ineptepub.py#L228

Don

**Comment by dmunsil** [comment]
Well, I dug deeper and discovered that for some reason, B&N doesn't put the rights.xml into the actual epub file for comic books. There's a copy of the rights.xml file stored in a SQLLite db in the Nook data directory. It's relatively straightforward to find the correct xml in that file, save it as "rights.xml", copy it into the epub file in the right place, and then calibre can decrypt the files. Unfortunately, the finished epub appears to be noncompliant in some way. All the page files are there inside the file in the "images" folder, but opening the file with any normal epub reader displays a basically empty file, with just a cover. I assume Nook must use a weird noncompliant ePub format for comics that they've never bothered to update.

**Reply by ElleKayEm** [reply]
I believe it's because they aren't standard epubs that any general epub reader can handle.  They only work in the nook app.  I have just extracted the images and created a cbz.  I left a comment on your other post with links where I went into detail on how I did it in the past.
